src/Codeception/Lib/Connector/Guzzle6.php

```diff
@@ -94,17 +94,17 @@ public function setAuth($username, $password)
     protected function createResponse(Psr7Response $response)
     {
         $body = (string) $response->getBody();
-        $headers = $this->flattenHeaders($response->getHeaders());
+        $headers = $response->getHeaders();
 
         $contentType = null;
-        if (isset($headers['Content-Type'])) {
-            $contentType = $headers['Content-Type'];
-        }
-        if (!$contentType or strpos($contentType, 'charset=') === false) {
-            if (preg_match('/\<meta[^\>]+charset *= *["\']?([a-zA-Z\-0-9]+)/i', $body, $matches)) {
-                $contentType .= ';charset=' . $matches[1];
+        if (isset($headers['Content-Type']) && count($headers['Content-Type']) === 1) {
+            $contentType = reset($headers['Content-Type']);
+            if (strpos($contentType, 'charset=') === false) {
+                if (preg_match('/\<meta[^\>]+charset *= *["\']?([a-zA-Z\-0-9]+)/i', $body, $matches)) {
+                    $contentType .= ';charset=' . $matches[1];
+                }
+                $headers['Content-Type'] = $contentType;
             }
-            $headers['Content-Type'] = $contentType;
         }
```

Comment:
I thried to fix the magic above to the best of my ability but I actually think it should be removed completely.